kafka-rest-common/src/main/java/io/confluent/kafkarest/KafkaExceptionMapper.java

```diff
@@ -0,0 +1,135 @@
+/*
+ * Copyright 2019 Confluent Inc.
+ *
+ * Licensed under the Confluent Community License (the "License"); you may not use
+ * this file except in compliance with the License.  You may obtain a copy of the
+ * License at
+ *
+ * http://www.confluent.io/confluent-community-license
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OF ANY KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations under the License.
+ */
+
+package io.confluent.kafkarest;
+
+import io.confluent.rest.RestConfig;
+import io.confluent.rest.entities.ErrorMessage;
+import io.confluent.rest.exceptions.GenericExceptionMapper;
+import org.apache.kafka.common.KafkaException;
+import org.apache.kafka.common.errors.*;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import javax.ws.rs.core.Response;
+import javax.ws.rs.core.Response.Status;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.concurrent.ExecutionException;
+
+public class KafkaExceptionMapper extends GenericExceptionMapper {
+
+  private static final Logger log = LoggerFactory.getLogger(KafkaExceptionMapper.class);
+
+  //Don't change the error codes, These are fixed for kafka-rest
+  public static final int KAFKA_BAD_REQUEST_ERROR_CODE = 40002;
+  public static final int KAFKA_AUTHENTICATION_ERROR_CODE = 40101;
+  public static final int KAFKA_AUTHORIZATION_ERROR_CODE = 40301;
+  public static final int TOPIC_NOT_FOUND_ERROR_CODE = 40401;
+  public static final int PARTITION_NOT_FOUND_ERROR_CODE = 40402;
+  public static final int KAFKA_UNKNOWN_TOPIC_PARTITION_CODE = 40403;
+  public static final int KAFKA_ERROR_ERROR_CODE = 50002;
+  public static final int KAFKA_RETRIABLE_ERROR_ERROR_CODE = 50003;
+  public static final int BROKER_NOT_AVAILABLE_ERROR_CODE = 50302;
+
+  public KafkaExceptionMapper(RestConfig restConfig) {
+    super(restConfig);
+  }
+
+  private static final Map<Class<? extends ApiException>, ResponsePair> HANDLED = errorMap();
```

Comment:
nit: Do you think a more descriptive name like `MAPPED_EXCEPTIONS` or `EXPECTED_EXCEPTIONS` would be more appropriate?